### ISSUE TITLE
md [hl]: header hightlighting spacing bug?

### ISSUE THREAD

**Issue by JohnVeness** [gsantner/markor#314]
#### General information

* **App version:** 1.0.0 F-Droid
* **System:** Android 6.0.1 on Xperia Z5C

#### Description
Many thanks for this app. I have been using it to convert some previously text-only documents I've created into Markdown documents.

However, I've found some rendering problems in the editor when Syntax highlighting is turned on. I haven't been able to boil this down to a simple example, but I attach a (redacted) typical file, and some screenshots of the problem. This is happening to almost any previously text-only file I try to convert to Markdown. Note that they render fine in preview mode, just not in the syntax highlighting editor.

Please note that the filename actually ends in .md - I only changed it to .txt so that github would accept it.
[example1.txt](https://github.com/gsantner/markor/files/2300360/example1.txt)
![screenshot_20180819-150426](https://user-images.githubusercontent.com/2512915/44309613-53846000-a3c1-11e8-9dd4-2604a412fd5f.png)
![screenshot_20180819-150439](https://user-images.githubusercontent.com/2512915/44309614-57b07d80-a3c1-11e8-87ce-9cbcc93fdfaa.png)
![screenshot_20180819-150447](https://user-images.githubusercontent.com/2512915/44309615-5aab6e00-a3c1-11e8-96d8-60461e77c095.png)
![screenshot_20180819-150457](https://user-images.githubusercontent.com/2512915/44309617-5e3ef500-a3c1-11e8-999e-5f5df6b77c40.png)

**Comment by gsantner** [comment]
Thanks for detailed report,!

Can you please try the previous markor version 0.3.10 and tell me if it is working better/worse/good there for you? There was a change for the Headers at 1.0.0, this is probably only a new problem for that.

**Comment by JohnVeness** [comment]
Unfortunately it's rendering exactly the same in 0.3.10. That version was the first version of your app I tried and where I first spotted the issue but didn't report it. I saw 1.0.0 and hoped this may have fixed the issue, especially with the note about headings, but I saw it was the same and thought it was about time I reported it!

Note that my workflow, when converting these previously text-only files to Markdown is generally just to add headings, i.e. ## in front of lines, and that is when things seem to go wrong.

If you would like a non-redacted version of the .md file, for easier reading, let me know a non-public way I can get it to you.

**Comment by gsantner** [comment]
No text doesnt matter, dont need other one - just one that does bad.
I currently cannot reproduce this on my devices, but will fix it when I get see it (or somebody else does)

**Comment by JohnVeness** [comment]
Unfortunately, this problem still seems to occur in version 1.1.1 downloaded from F-Droid today.

**Comment by gsantner** [comment]
set line spacing in settings to 100% or more.

**Comment by JohnVeness** [comment]
Ah, I saw you say that earlier but I noticed the commit and I thought you were saying that the commit would change the line spacing automatically. I see now that this is an option we need to change!

Unfortunately, the problem still occurs with 100. Still occurs with 120, although less overlap. Have to go to 150 for no overlap, but that means I'm not seeing much on the screen! Even with 150 I can see where the problem occurs, where lines are closer than normal.
